### PR TITLE
Make Document::DefaultView return a null value when there's no browsing context

### LINKED ISSUE #11469: Make Document::DefaultView return a null value when there's no browsing context
This requires updating `Document.webidl` to indicate that a [nullable return value](http://heycam.github.io/webidl/#idl-nullable-type) is allowed, then modifying the implementation to check `self.browsing_context()` and return an appropriate value.

**Code**: `components/script/dom/webidls/Document.webidl`, `components/script/dom/document.rs`
**Test**: `./mach test-wpt tests/wpt/web-platform-tests/html/browsers/the-window-object/Document-defaultView.html` should show newly-passing tests, so the expected results will need to be [updated](https://github.com/servo/servo/blob/master/tests/wpt/README.md#updating-test-expectations).

### PR DESCRIPTION
**Pull request by kevgs**
Fixes #11469.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11548)

### CODE REVIEW

File: tests/wpt/web-platform-tests/cssom-view/scrolling-no-browsing-context.html [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<!doctype html>
+<meta charset="utf-8">
+<title>cssom-view scrolling-no-browsing-context</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+setup({explicit_done:true});
+window.onload = function () {
+        test(function () {
+            doc = document.implementation.createDocument("http://example.com/", "html", null);
```
Comment (Ms2ger) [inline tests/wpt/web-platform-tests/cssom-view/scrolling-no-browsing-context.html]: Should be `var doc`

File: tests/wpt/web-platform-tests/cssom-view/scrolling-no-browsing-context.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<!doctype html>
+<meta charset="utf-8">
+<title>cssom-view scrolling-no-browsing-context</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+setup({explicit_done:true});
+window.onload = function () {
+        test(function () {
```
Comment (Ms2ger) [inline tests/wpt/web-platform-tests/cssom-view/scrolling-no-browsing-context.html]: This whole block is indented four spaces too far.

**Comment by Ms2ger** [comment]
Two nits for the test, LGTM otherwise.

**Comment by Ms2ger** [comment]
Please squash the commits; let me know or ask on IRC if you need help with that.

**Comment by kevgs** [comment]
Thanks for your time :)

**Comment by Ms2ger** [comment]
@bors-servo retry
* network?

**Comment by jdm** [comment]
The dom/nodes/Node-parentNode.html failure is #11613, but the other unexpected results need to be investigated.

**Comment by jdm** [comment]
Travis is complaining that the manifest changes need to be reordered. This can be fixed with `./mach update-manifest`.
```
$ bash etc/ci/manifest_changed.sh
diff --git a/tests/wpt/metadata/MANIFEST.json b/tests/wpt/metadata/MANIFEST.json
index a15017e..c617116 100644
--- a/tests/wpt/metadata/MANIFEST.json
+++ b/tests/wpt/metadata/MANIFEST.json
@@ -36030,17 +36030,17 @@
     "deleted_reftests": {},
     "items": {
       "testharness": {
-        "url/url-domainToUnicode.html": [
-          {
-            "path": "url/url-domainToUnicode.html",
-            "url": "/url/url-domainToUnicode.html"
-          }
-        ],
         "cssom-view/scrolling-no-browsing-context.html": [
           {
             "path": "cssom-view/scrolling-no-browsing-context.html",
             "url": "/cssom-view/scrolling-no-browsing-context.html"
           }
+        ],
+        "url/url-domainToUnicode.html": [
+          {
+            "path": "url/url-domainToUnicode.html",
+            "url": "/url/url-domainToUnicode.html"
+          }
         ]
       }
     },
```

File: components/script/dom/webidls/Document.webidl [hunk 3]
```diff
@@ -118,7 +118,7 @@ partial /*sealed*/ interface Document {
   // void writeln(DOMString... text);
 
   // user interaction
-  readonly attribute Window/*Proxy?*/ defaultView;
+  readonly attribute Window?/*Proxy?*/ defaultView;
```
Comment (KiChjang) [inline components/script/dom/webidls/Document.webidl]: Why are you changing this?

Comment (kevgs) [inline components/script/dom/webidls/Document.webidl]: It was required here https://github.com/servo/servo/issues/11469

**Comment by KiChjang** [comment]
@kevgs Does the latest commit you pushed fix the timeouts encountered in the test files?

**Comment by kevgs** [comment]
No. I just hope there will be no timeouts. And i'll fix them if the still present.

**Comment by KiChjang** [comment]
Can you try running those tests locally and see whether they timeout? (`./mach test-wpt tests/wpt/web-platform-tests/dom/nodes/Document-create*`)

**Comment by kevgs** [comment]
Tests worked without timeouts in release mode.

**Comment by jdm** [comment]
@bors-servo: retry
* #11613

**Pull request merged**